### PR TITLE
Integrate pandas matplotlib, and template for AIL and FT

### PR DESCRIPTION
**Pull request by bobbyc**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -13,8 +13,14 @@ argparse==1.2.1
...
 peakutils==1.0.3
 treeherder-client==3.1.0
+<<<<<<< HEAD
+matplotlib>=1.5.3
+pandas>=0.19.2
+
+=======
 geckoprofiler_controller==0.0.4
 # matplotlib>=1.5.1
+>>>>>>> master
```
Comment (MikeLien) [inline requirements.txt]: please fix the conflict here

File: requirements.txt [hunk 2]
```diff
@@ -13,8 +13,14 @@ argparse==1.2.1
 watchdog==0.8.3
 peakutils==1.0.3
 treeherder-client==3.1.0
+<<<<<<< HEAD
+matplotlib>=1.5.3
+pandas>=0.19.2
+
+=======
```
Comment (ypwalter) [inline requirements.txt]: should solve this conflict

File: tools/FrameThroughputTemplate.ipynb [hunk 3]
```diff
@@ -0,0 +1,353 @@
+{
```
Comment (MikeLien) [inline tools/FrameThroughputTemplate.ipynb]: suggest aggregate *.ipynb into one folder for their specific purpose

File: tools/InputLatency.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            runtime = pd.DataFrame([pd.DataFrame(d[c]['time_list'])['run_time'] for c in d]).T
+            runtime.columns = d.columns
+            print '===== Input Latency Raw data===='
+            print runtime
+            print '\n\n===== Input Latency Percentile===='
+            print runtime.quantile([0.01, 0.99], interpolation='nearest')
+            print '\n\n===== Input Latency Summary Table===='
+            print runtime.describe()
+            print '\n\n===== Input Latency Plots===='            
+            runtime.plot.hist(bins=20, histtype='bar', color=['green','orange'])
```
Comment (askeing) [inline tools/InputLatency.py]: base on PEP8, please add a whitespace before `'orange'`.

File: tools/InputLatency.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            runtime.plot.box()
+            #plt.show()
```
Comment (askeing) [inline tools/InputLatency.py]: removing this unused code, or starting the comment with `# `.

File: tools/InputLatency.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+import pandas as pd
+import numpy as np
```
Comment (askeing) [inline tools/InputLatency.py]: `./tools/InputLatency.py:4: F401 'np' imported but unused`

File: tools/InputLatency.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    def run(self):
+        """
+        run()
+        """
+        self.generate_metric()
+
+
+def main():
+    """ 
+    function main() as entry point
```
Comment (askeing) [inline tools/InputLatency.py]:
```
./tools/InputLatency.py:70: trailing whitespace
./tools/InputLatency.py:71: trailing whitespace
```

Comment (askeing) [inline tools/InputLatency.py]: you can also run `./mach test-tidy --all` to test the code locally.

File: tools/InputLatency.py (hunk 6 above)
Comment (bobbyc) [inline tools/InputLatency.py]: It's ok. Just put there as template combination. Pandas, numpy, and matplotlib always works together

File: tools/InputLatency.py (hunk 7 above)
Comment (bobbyc) [inline tools/InputLatency.py]: what is this?

Comment (askeing) [inline tools/InputLatency.py]: source code checker, like flake8

**Pull request closed without merging**